oscap_report/old_html_report_style/report_generator.py

```diff
@@ -1,7 +1,26 @@
+from io import BytesIO
+from pathlib import Path
+
+from lxml import etree
+
+XSL_DIR = Path(__file__).parent / "xsl"
+
 
 class ReportGenerator():
-    def __init__(self, data):
-        self.data = data
+    def __init__(self, parser):
+        self.xml_report = parser.root
+        self.xslt_doc = etree.parse(str(XSL_DIR / "xccdf-report.xsl"))
+        self.xslt_transformer = etree.XSLT(self.xslt_doc)
 
     def generate_html_report(self):
-        return "<html><h1>TEST REPORT</h1><h2>{}</h2></html>\n".format(self.data)
+        html_report = self.xslt_transformer(self.xml_report)
+        result_html = etree.tostring(
+            html_report,
+            xml_declaration=True,
+            doctype=("<!DOCTYPE html>"),
```

Comment:
I'm not sure what you were trying to do here, but the result of `("<!DOCTYPE html>"),` is actually `('<!DOCTYPE html>',)`. I'd fix it even if only just to be explicit about means.

Reply:
It is a mistake. I used an older version of doctype that is longer. I use parentheses to format a long string. I removed it.